lua/starfall/libs_sh/vr.lua

```diff
@@ -33,192 +21,158 @@ SF.hookAdd("VRMod_Start", "vrstart")
 -- @param ply Player exiting VR
 SF.hookAdd("VRMod_Exit", "vrexit")
 
+if CLIENT then
+
+--- This gets called every time a boolean controller input action changes state
+-- @name VRInput
+-- @class hook
+-- @param actionname Name of the input
+-- @param boolean State of the input
+-- @client
+SF.hookAdd("VRMod_Input", "vrinput")
+
+--- Called before rendering the game. Any code that you want to run once per frame should be put here. HUD is required.
+-- @name VRPreRender
+-- @class hook
+-- @client
+SF.hookAdd("VRMod_PreRender", "vrprerender", canRenderHudSafeArgs)
+
+--- Called before rendering the right eye. This along with the previous hook can be used to render different things in different eyes. HUD is required.
+-- @name VRPreRenderRight
+-- @class hook
+-- @client
+SF.hookAdd("VRMod_PreRenderRight", "vrprerenderright", canRenderHudSafeArgs)
+
+--- Called after rendering the game. HUD is required.
+-- @name VRPostRender
+-- @class hook
+-- @client
+SF.hookAdd("VRMod_PostRender", "vrpostrender", canRenderHudSafeArgs)
+
+end 
+
+return function(instance)
+
+if not vrmod return end --only add these functions if vrmod is installed
+
+local vr_library = instance.Libraries.vr
+
+local owrap, ounwrap = instance.WrapObject, instance.UnwrapObject
+local ents_methods, ent_meta, ewrap, eunwrap = instance.Types.Entity.Methods, instance.Types.Entity, instance.Types.Entity.Wrap, instance.Types.Entity.Unwrap
+local ang_meta, awrap, aunwrap = instance.Types.Angle, instance.Types.Angle.Wrap, instance.Types.Angle.Unwrap
+local vec_meta, vwrap, vunwrap = instance.Types.Vector, instance.Types.Vector.Wrap, instance.Types.Vector.Unwrap
+local plywrap = instance.Types.Player.Wrap
+
+local getply
+instance:AddHook("initialize", function()
+	getply = instance.Types.Player.GetPlayer
+end)
 
 --- Checks wether the player is in VR
--- @class function
 -- @param target player to check
 -- @return boolean true if player is in VR
 function vr_library.isPlayerInVR(ply)
-	local ply = plyunwrap(ply)
-	checkluatype(ply, TYPE_ENTITY) 
+	local ply = getply(ply)
```

Comment:
You can move all the getply(ply) inline inside the function using it now instead of creating a local.

Reply:
So i can just pass the ply like that?

Reply:
Yah. `return vrmod.IsPlayerInVR(getply(ply))`